django/test/testcases.py

```diff
@@ -1071,6 +1073,55 @@ def connections_support_transactions(aliases=None):
     return all(conn.features.supports_transactions for conn in conns)
 
 
+class TestData:
+    """
+    Descriptor meant to provide TestCase instance isolation for attributes
+    assigned during the setUpTestData phase.
+
+    It allows safe alteration of objects assigned in setUpTestData by test
+    methods by exposing deep copies instead of the original objects.
```

Comment:
Might also extend this comment just to point out that memo is shared per instance to allow references between copied objects to carry over.